purposes-archiver/src/main/scala/it/pagopa/interop/purposesarchiver/Main.scala

```diff
@@ -0,0 +1,90 @@
+import it.pagopa.interop.purposesarchiver.system.ApplicationConfiguration
+import it.pagopa.interop.purposesarchiver.service.Dependencies
+import it.pagopa.interop.purposesarchiver.service.impl.QueueServiceImpl
+import it.pagopa.interop.purposeprocess.client.model.PurposeVersionState._
+import it.pagopa.interop.agreementprocess.events.ArchiveEvent
+import it.pagopa.interop.purposeprocess.client.model.Purpose
+import it.pagopa.interop.commons.utils.BEARER
+import scala.concurrent.{ExecutionContext, Future, Await, ExecutionContextExecutor}
+import scala.concurrent.duration.Duration
+import scala.util.{Failure, Success}
+import java.util.concurrent.{ExecutorService, Executors}
+
+case object ErrorShutdown   extends CoordinatedShutdown.Reason
+case object SuccessShutdown extends CoordinatedShutdown.Reason
+
+object Main extends App with Dependencies {
+
+  implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  implicit val actorSystem: ActorSystem[Nothing] = ActorSystem[Nothing](Behaviors.empty, "interop-be-purposes-archiver")
+  implicit val executionContext: ExecutionContext      = actorSystem.executionContext
+  implicit val classicActorSystem: classic.ActorSystem = actorSystem.toClassic
+
+  val blockingThreadPool: ExecutorService = Executors.newFixedThreadPool(Runtime.getRuntime.availableProcessors())
+  implicit val blockingEC: ExecutionContextExecutor = ExecutionContext.fromExecutor(blockingThreadPool)
+
+  val queueService = new QueueServiceImpl(
+    ApplicationConfiguration.purposesArchiverQueueName,
+    ApplicationConfiguration.visibilityTimeoutInSeconds
+  )
+
+  val ARCHIVABLE_STATES = Seq(ACTIVE, SUSPENDED, WAITING_FOR_APPROVAL, DRAFT)
+
+  private def processVersion(purpose: Purpose)(implicit contexts: Seq[(String, String)]): Future[Unit] =
+    purpose.versions.maxByOption(_.createdAt) match {
+      case Some(x) =>
+        x.state match {
+          case ACTIVE | SUSPENDED           => purposeProcessService.archive(purpose.id, x.id).map(_ => ())
+          case DRAFT | WAITING_FOR_APPROVAL => purposeProcessService.delete(purpose.id, x.id)
+          case _                            => Future.unit
+        }
+      case None    => Future.unit
+    }
+
+  private def execution: Future[Unit] = queueService.processMessages { message =>
+    for {
+      bearer <- generateBearer
+      _                 = logger.info("Internal Token obtained")
+      contextWithBearer = contexts :+ (BEARER -> bearer)
+      agreement <- agreementProcessService.getAgreementById {
+        message.payload match {
+          case m: ArchiveEvent => m.agreementId
+          case other           => {
+            logger.error("Purposes archiver job failed because message is not compliant")
+            throw EventNotComplaint(other.getClass().toString)
+          }
+        }
+      }(contextWithBearer)
```

Comment:
```suggestion
      agreementId <- message.payload match {
          case m: ArchiveEvent => Future.succesful(m.agreementId)
          case _ => 
            logger.error("Purposes archiver job failed because message is not compliant")
            Future.failed(EventNotComplaint(other.getClass().toString))
      agreement <- agreementProcessService.getAgreementById(agreementId)(contextWithBearer)
```